### PR TITLE
Removed dependency on VS Telemetry.

### PR DESCRIPTION
**Pull request by harshjain2**
Removed dependency on VS Telemetry.

### CODE REVIEW

File: src/vstest.console/Publisher/MetricsPublisherFactory.cs [hunk 1]
```diff
@@ -18,11 +18,6 @@ public class MetricsPublisherFactory
         /// <returns>Returns Instance of Metrics Publisher</returns>
         public static async Task<IMetricsPublisher> GetMetricsPublisher(bool isTelemetryOptedIn, bool isDesignMode)
         {
-            if (isTelemetryOptedIn && !isDesignMode)
-            {
-                return await Task.Run(() => new MetricsPublisher());
-            }
-
             return await Task.Run(() => new NoOpMetricsPublisher());
```
Comment (codito) [inline src/vstest.console/Publisher/MetricsPublisherFactory.cs]: Please use `Task.FromResult`

File: src/vstest.console/Publisher/NoOpMetricsPublisher.cs [hunk 2]
```diff
@@ -17,7 +23,12 @@ public class NoOpMetricsPublisher : IMetricsPublisher
         /// <param name="metrics"></param>
         public void PublishMetrics(string eventName, IDictionary<string, object> metrics)
         {
-            // No Operation
+            // Log to Text File
+            var logEnabled = Environment.GetEnvironmentVariable("VSTEST_LOGTELEMETRY");
+            if (!string.IsNullOrEmpty(logEnabled) && logEnabled.Equals("1", StringComparison.Ordinal))
+            {
+                this.LogToFile(eventName, metrics, new FileHelper());
```
Comment (codito) [inline src/vstest.console/Publisher/NoOpMetricsPublisher.cs]: Suggest to add a console output or trace the file name to allow correlation.

File: src/vstest.console/Publisher/NoOpMetricsPublisher.cs [hunk 3]
```diff
@@ -17,7 +23,12 @@ public class NoOpMetricsPublisher : IMetricsPublisher
         /// <param name="metrics"></param>
         public void PublishMetrics(string eventName, IDictionary<string, object> metrics)
         {
-            // No Operation
+            // Log to Text File
+            var logEnabled = Environment.GetEnvironmentVariable("VSTEST_LOGTELEMETRY");
```
Comment (codito) [inline src/vstest.console/Publisher/NoOpMetricsPublisher.cs]: Should this check always happen even if telemetry is opted out?

File: src/vstest.console/Publisher/NoOpMetricsPublisher.cs (hunk 2 above)
Comment (harshjain2) [reply src/vstest.console/Publisher/NoOpMetricsPublisher.cs]: Can we log the file in EqtTrace logs?

**Review by codito — COMMENTED** [review]
Few suggestions

**Review by codito — APPROVED** [review]

**Pull request merged**